extensions/stackdriver/stackdriver.h

```diff
@@ -0,0 +1,83 @@
+/* Copyright 2019 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef STACKDRIVER_H
+#define STACKDRIVER_H
+
+#include "opencensus/exporters/stats/stackdriver/stackdriver_exporter.h"
+
+#ifndef NULL_PLUGIN
+#include "api/wasm/cpp/proxy_wasm_intrinsics.h"
+#else
+
+#include "extensions/common/wasm/null/null.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace Common {
+namespace Wasm {
+namespace Null {
+namespace Plugin {
+using namespace Plugin;
+#endif
+
+namespace Stackdriver {
+
+#ifdef NULL_PLUGIN
+NULL_PLUGIN_ROOT_REGISTRY;
+#endif
+
+// StackdriverRootContext is the root context for all streams processed by the
+// thread. It has the same lifetime as the worker thread and acts as target for
+// interactions that outlives individual stream, e.g. timer, async calls.
+class StackdriverRootContext : public RootContext {
+ public:
+  StackdriverRootContext(uint32_t id, StringView root_id)
```

Comment:
Question: What is the full namespace of StringView?

Reply:
In native mode, StringView = absl::string_view, because IIRC envoy is using C++ 14 and string_view is not available in stdlib:
https://github.com/envoyproxy/envoy-wasm/blob/f66e557b6e787043e9efde5cfcd88afdce0b9408/source/extensions/common/wasm/null/null.h#L20

In wasm mode, StringView = std::string_view: https://github.com/envoyproxy/envoy-wasm/blob/f66e557b6e787043e9efde5cfcd88afdce0b9408/api/wasm/cpp/proxy_wasm_intrinsics.h#L15